File: Assignment 3/Sort.py
'''
    Name: Josh Moore
    Date: 21 September 2023
    Assignment: Sorting
    Description: Intake a user input of number array, then sort into ascending order.
'''

# Variables
array = [];

'''
    Type: Function
    Name: getUserInput
    Return: Void
    Description: Gets the user's input. in total, asks the user to input ten numerical values in any order. If the
                 inputted value is not a numerical value, then the loop exists.
'''
def getUserInput():
    userInput = input("Enter a number (Use non-number to stop): ");
    if(userInput.isdigit()):
            array.append(int(userInput));

    while userInput.isdigit() and len(array) < 10:
        userInput = input("Enter a number (Use non-number to stop): ");

        if(userInput.isdigit()):
            array.append(int(userInput));

'''
    Type: Function
    Name: sort
    Return: Void
    Description: Sorts through the array of user inputted values. The sorting method is selection-sort, meaning it looks at each
                 value past a current value in the array, to check if the value is less than the current value. This sorting method
                 takes a complexity of O(n).
'''
def sort():
    for i in range(len(array)):
        for j in range(i + 1, len(array)):
            if array[j] < array[i]:
                tmp = array[i];
                array[i] = array[j];
                array[j] = tmp;

# Main call functions
getUserInput();
sort();
print(array);

File: Assignment 3/Sort.pl
# Variables
my @array = ();

# Function Calling
userInput(); # Gets User input
sortArray(); # Sorts the array.

# Prints out array
print("@array");

=begin comment
    Type: Function
    Name: userInput
    Return: Void
    Description: Asks the ten times to enter in Integer Values. Once the array has reached
    the maximum of ten values, the function is exited.
=end comment
=cut
sub userInput {
    while(scalar @array < 10) {
        print("Enter in value: ");
        my $tmp = <STDIN>;
        push(@array, $tmp);
    }

    return undef
}

=begin comment
    Type: function
    Name: sortArray
    Return: Void
    Description: Sorts the array in ascending order. If the OUTTER value is greater than the INNER value, then
    the two values are switched in placement of the array.
=end comment
=cut
sub sortArray {
=multi-line comment
    for(my $i = 0; $i < scalar @array; $i++) {
        for(my $j = $i + 1; $j < scalar @array; $j++) {
            if($array[j] < $array[i]) {
                my $tmp = $array[i];
                $array[i] = $array[j];
                $array[j] = $tmp;
            }
        }
    }
=end comment
=cut

    foreach $i (@array) {
        foreach $j (@array) {
            if($j > $i) {
                ($i, $j) = ($j, $i);
            }
        }
    }

    return undef
}